docusaurus/docs/iOS/guides/working-with-the-composer.md

```diff
@@ -0,0 +1,221 @@
+---
+title: Working with the Composer
+---
+
+The `ComposerVC` is the component responsible to create new messages or change existing messages. The composer belongs to the `ChatVC` component that integrates the message list and the composer in one view. But if you are not using our `ChatVC` you can use the composer component in isolation with your own chat view as well. The `ComposerVC` manages all the logic and actions of the `ComposerView`, where the latter is only responsible for the composer's layout.
+
+## Customizing the Composer UI
+By subclassing the `ComposerView` you can customize the layout of the Composer, for example, changing the position of the views, removing views or adding new ones. Since all the logic of the composer lives on the `ComposerVC`, this is the class to override in case the customization depends on data. Here is an example of customizing the composer to look like the iMessage app:
+```swift
+final class iMessageComposerView: ComposerView {
+    override func setUpLayout() {
+        super.setUpLayout()
+
+        leadingContainer.removeArrangedSubview(commandsButton)
+        trailingContainer.removeArrangedSubview(sendButton)
+        inputMessageView.inputTextContainer.addArrangedSubview(sendButton)
+    }
+    
+    override func setUpAppearance() {
+        super.setUpAppearance()
+
+        attachmentButton.setImage(
+            UIImage(systemName: "camera.fill"),
+            for: .normal
+        )
+    }
+}
+
+final class iMessageComposerVC: ComposerVC {
+    override func updateContent() {
+        super.updateContent()
+
+        switch content.state {
+        case .new:
+            composerView.inputMessageView.textView.placeholderLabel.text = "iMessage"
+        default:
+            break
+        }
+    }
+}
```

Comment:
We should add comments to this code and explain why we're doing the stuff we're doing